src/components/ProfileCard.jsx

```diff
@@ -1,84 +1,91 @@
-import React from "react";
-import profileImg from "../assets/images/profile-image.png";
-import messages from "../assets/icons/Messages.svg";
+import React, { useState, useEffect } from "react";
 import styled from "styled-components";
+import ProfileCardItem from "./ProfilecardItem";
+import Pagination from "./Pagination";
 
-const Card = styled.div`
-  height: 187px;
-  border-radius: 12px;
-  box-shadow: 0px 4px 12px rgba(0, 0, 0, 0.1);
+const ProfileContainer = styled.div`
   display: flex;
   flex-direction: column;
-  align-items: center;
-  justify-content: space-around;
-  background-color: white;
-  border: 1px solid var(--Grayscale-40);
-  cursor: pointer;
+  gap: 45px;
+  max-width: 1200px;
+  width: 90%;
+  margin: 0 auto;
 `;
 
-const UserNameContainer = styled.div`
-  width: 180px;
-  height: 97px;
-  display: flex;
-  flex-direction: column;
-  align-items: flex-start;
-`;
+const ProfileCardContainer = styled.div`
+  display: grid;
+  grid-template-columns: repeat(4, minmax(186px, 220px));
+  gap: 2rem;
+  justify-content: center;
 
-const ProfileImage = styled.img`
-  width: 60px;
-  height: 60px;
-  border-radius: 50%;
-  object-fit: cover;
-  margin-bottom: 15px;
+  @media screen and (max-width: 865px) {
+    grid-template-columns: repeat(3, minmax(186px, 220px));
+  }
+  @media screen and (max-width: 600px) {
+    grid-template-columns: repeat(2, minmax(155px, 220px));
+    gap: 1.6rem;
+  }
 `;
 
-const Username = styled.h3`
-  font-size: 20px;
-  font-weight: 400;
-`;
+function ProfileCard({ profiles }) {
+  const [currentPage, setCurrentPage] = useState(1); // 현재 페이지 상태
+  const [cardsPerPage, setCardsPerPage] = useState(8); // 기본적으로 8개 카드 표시
 
-const QuestionContainer = styled.div`
-  width: 80%;
-  display: flex;
-  flex-direction: row;
-  justify-content: space-between;
-  align-items: center;
-  color: var(--Grayscale-40);
-  font-size: 14px;
-`;
+  // 화면 크기에 따라 페이지당 카드 수를 동적으로 조정
+  useEffect(() => {
+    const handleResize = () => {
+      if (window.innerWidth <= 865) {
+        setCardsPerPage(6); // 화면 너비가 865px 이하일 경우 6개 카드로 변경
+      } else {
+        setCardsPerPage(8); // 기본적으로 8개 카드 표시
+      }
+    };
 
-const QuestionIcon = styled.span`
-  display: flex;
-  align-items: center;
+    handleResize(); // 초기 로드 시 체크
+    window.addEventListener("resize", handleResize); // 리사이즈 시 카드 수 업데이트
 
-  > img {
-    width: 20px;
-    height: 20px;
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, []);
 
-    opacity: 0.6;
-  }
-`;
+  // 프로필 데이터를 `cardsPerPage`만큼 나누는 함수
+  const groupProfiles = (profilesArray, groupSize) => {
+    const grouped = [];
```

Comment:
(개인 스타일이니 참고)
배열 요소가 없으면 앞단에서 실행을 종료하는게 더 효율적일것 같아요~
```suggestion
    if (!profilesArray || profilesArray.length === 0) {
      return [];
    }
    const grouped = [];
```